### PR TITLE
batches: UI papercut fixes

### PR DESCRIPTION
**Pull request by st0nebraker**
1. GitHub App logo error fallback image
**BEFORE**
image

**AFTER**
image

2. GitHub App Edit screen mobile responsiveness (Down to 320px screenwidth)
**BEFORE**
image

**AFTER**
image

## Test plan
- Manual visual testing

### CODE REVIEW

File: client/web/src/components/gitHubApps/AppLogo.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            className={className}
+            src={src}
+            alt="App logo"
+            aria-hidden={true}
+            onError={() => {
+                setFallbackImage(true)
+            }}
+        />
+    ) : (
+        <UserAvatar user={{ avatarURL: null, username: name, displayName: name }} className={className} />
```
Comment (courier-new) [inline client/web/src/components/gitHubApps/AppLogo.tsx:25]: Just confirming, there's nothing actually user-specific to the implementation of this component in the HTML that gets rendered, right? Like it's not aria-labeled as "Avatar for user X" or given an id/class related to a user or anything, right?

File: client/web/src/components/gitHubApps/AppLogo.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import { UserAvatar } from '@sourcegraph/shared/src/components/UserAvatar'
+
+interface AppLogoProps {
+    src: string | undefined
+    name: string
+    className: string
+}
+
+export const AppLogo: React.FC<AppLogoProps> = ({ src, name, className }) => {
```
Comment (courier-new) [inline client/web/src/components/gitHubApps/AppLogo.tsx:11]: Do you mind adding a docstring just to explain why this component exists, for posterity? 🙂

File: client/web/src/components/gitHubApps/AppLogo.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    return !fallbackImage ? (
+        <img
+            className={className}
+            src={src}
+            alt="App logo"
+            aria-hidden={true}
+            onError={() => {
+                setFallbackImage(true)
+            }}
```
Comment (courier-new) [inline client/web/src/components/gitHubApps/AppLogo.tsx:22]: Maybe also an inline comment here?

```suggestion
			// On certain code hosts, the image resource may be locked behind a login
			// screen. It's not practical to authenticate the user in this context, so instead,
			// we catch when there's an error loading the image and toggle the component
			// state to render a fallback icon instead.
            onError={() => setFallbackImage(true)}
```

File: client/web/src/components/gitHubApps/GitHubAppCard.tsx [hunk 4]
```diff
@@ -33,7 +34,7 @@ export const GitHubAppCard: React.FC<GitHubAppCardProps> = ({ app, refetch }) =>
                 <RemoveGitHubAppModal onCancel={() => setRemoveModalOpen(false)} afterDelete={refetch} app={app} />
             )}
             <span className={classNames(styles.appLink, 'd-flex align-items-center')}>
-                <img className={classNames('mr-3', styles.logo)} src={app.logo} alt="app logo" aria-hidden={true} />
```
Comment (courier-new) [inline client/web/src/components/gitHubApps/GitHubAppCard.tsx:36]: Would it be possible to preserve the alt text as well?

**Review by courier-new — APPROVED** [review]
Thanks for fixing this up. 🙏

File: client/web/src/components/gitHubApps/AppLogo.tsx (hunk 1 above)
Comment (pjlast) [reply client/web/src/components/gitHubApps/AppLogo.tsx:25]: From the looks of it, the UserAvatar component has `aria-hidden={true}` set

**Review by pjlast — APPROVED** [review]
Thank you!

File: client/web/src/components/gitHubApps/AppLogo.tsx (hunk 1 above)
Comment (st0nebraker) [reply client/web/src/components/gitHubApps/AppLogo.tsx:27]: Correct, no specific `alt` text on `UserAvatar` and `aria-hidden={true}`

File: client/web/src/components/gitHubApps/GitHubAppCard.tsx (hunk 4 above)
Comment (st0nebraker) [reply client/web/src/components/gitHubApps/GitHubAppCard.tsx:36]: I have the alt text already preserved, but made it dynamic for the GH App installation use case [here](https://github.com/sourcegraph/sourcegraph/pull/53974/commits/64b8ce158bbb0db3505a4d4ec144b10338397d2f)! Let me know what you think

File: client/web/src/components/gitHubApps/AppLogo.tsx (hunk 2 above)
Comment (st0nebraker) [reply client/web/src/components/gitHubApps/AppLogo.tsx]: Done [here](https://github.com/sourcegraph/sourcegraph/pull/53974/commits/64b8ce158bbb0db3505a4d4ec144b10338397d2f)!

**Pull request merged**